packages/connection/src/class-plugin-storage.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Storage for plugin connection information.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection;
+
+/**
+ * Storage for plugin connection information.
+ *
+ * @todo Adapt for multisite installations.
+ */
+class Plugin_Storage {
+
+	const OPTION_KEY = 'connection_plugins';
+
+	/**
+	 * Connected plugins.
+	 *
+	 * @var array
+	 */
+	private static $plugins = array();
+
+	/**
+	 * Whether the plugins were configured.
+	 * To make sure we don't call the configuration process again and again.
+	 *
+	 * @var bool
+	 */
+	private static $plugins_configuration_finished = false;
+
+	/**
+	 * Add or update the plugin information in the storage.
+	 *
+	 * @param string $slug Plugin slug.
+	 * @param array  $args Plugin arguments, optional.
+	 *
+	 * @return bool
+	 */
+	public static function upsert( $slug, array $args = array() ) {
+		self::$plugins[ $slug ] = $args;
+
+		return true;
+	}
+
+	/**
+	 * Retrieve the plugin information by slug.
+	 *
+	 * @param string $slug The plugin slug.
+	 *
+	 * @return array|null
+	 */
+	public static function get_one( $slug ) {
+		return empty( self::$plugins[ $slug ] ) ? null : self::$plugins[ $slug ];
+	}
+
+	/**
+	 * Retrieve info for all plugins that use the connection.
+	 *
+	 * @return array
+	 */
+	public static function get_all() {
+		if ( ! self::$plugins_configuration_finished ) {
+			do_action( 'jetpack_connection_configure_plugin' );
```

Comment:
When adding hooks (`apply_filters`, `do_action`) to Jetpack files, we try to add a docblock to them every time. This is because on each release day, a parser looks at all files and builds / updates Codex pages for each Jetpack hook here:
https://developer.jetpack.com/

Here is an example:
https://github.com/Automattic/jetpack/blob/3010b47431d784ba4771188ca1da2637bb922d97/packages/connection/src/class-manager.php#L707-L714

Reply:
Good point, fixed now.